node/node.go

```diff
@@ -291,237 +186,6 @@ func readGenesisCid(ds datastore.Datastore) (cid.Cid, error) {
 	return c, nil
 }
 
-// buildHost determines if we are publically dialable.  If so use public
-// Address, if not configure node to announce relay address.
-func (nc *Config) buildHost(ctx context.Context, makeDHT func(host host.Host) (routing.Routing, error)) (host.Host, error) {
```

Comment:
This is too much networking logic for a builder. It should be a free function in a different file that takes `isRelay`, `publicRelayAddress`, and `libP2POptions` as parameters.

This could wait for later though.